### PR TITLE
Add equivalences from `CR<Pauli>` to `R<Paulis>`

### PR DESCRIPTION
**Pull request by Cryoris**
### Summary

Add equivalences from controlled Pauli rotations to 2-Pauli rotations. This is particularly interesting e.g. for QPD applications where cuts are usually defined for 2-Pauli rotations or for compiling to the RZX gate and using a pulse-efficient decomposition. This could e.g. be used to compile a `CRY` gate to an efficient pulse sequence instead of 2 CX gates.

### CODE REVIEW

**Comment by mtreinish** [comment]
It's not really related to this PR, as a standalone these equivalences are fine on their own. But one thing we are missing in the equivalence lib (it came up on [slack](https://qiskit.slack.com/archives/C034SHPSKSQ/p1674572813935989)) is targeting a `CR<Pauli>` basis. I think to fix it we can just define a rules from cx to the crx, etc to fix that.

**Review by jakelishman — APPROVED** [review]
(For posterity: the correctness of the standard equivalence library is tested automatically, without needing to modify the suite.)

File: qiskit/circuit/library/standard_gates/equivalence_library.py [hunk 1]
```diff
@@ -257,6 +284,20 @@
...
+# q_0: ┤0        ├   q_0: ┤ H ├┤0        ├┤ H ├
+#      │  Rxx(ϴ) │ ≡      └───┘│  Rzx(ϴ) │└───┘
+# q_1: ┤1        ├   q_1: ─────┤1        ├─────
+#      └─────────┘             └─────────┘
+theta = Parameter("theta")
+rxx_to_rzx = QuantumCircuit(2)
+rxx_to_rzx.h(0)
+rxx_to_rzx.rzx(theta, 0, 1)
+rxx_to_rzx.h(0)
+_sel.add_equivalence(RXXGate(theta), rxx_to_rzx)
```
[the remark this answers is not in the document]
Comment (Cryoris) [reply qiskit/circuit/library/standard_gates/equivalence_library.py:298]: Tbh I never know which order it is and just end up trying until it fits 😅 but I agree that `RZX` should implement `e^{i thate * ZX}` and in our tensor order that would be `Z` on `q_1`..

[the remark this answers is not in the document]
Comment (jakelishman) [reply qiskit/circuit/library/standard_gates/equivalence_library.py:298]: I think the argument is that it implements what our documentation calls :math:`Z \otimes X`, where I think there's some conflation of the general tensor product :math:`\otimes` with the Kronecker product ordering we use to make concrete matrix representations.

But anyway, the ship's sailed, and it's unrelated to this PR!

[the remark this answers is not in the document]
Comment (jakelishman) [reply qiskit/circuit/library/standard_gates/equivalence_library.py:298]: Ah - Wikipedia explains:

> the Kronecker product, sometimes denoted by $\otimes$

as opposed to:

> the tensor product $V\otimes W$

I've never used `\otimes` to refer to the Kronecker product, and wasn't aware that was something people did - to me, it's always meant the arbitrary tensor product, which isn't tied to any particular representation of linear algebra.

**Comment by Cryoris** [comment]
@mtreinish I opened an issue to keep track of these missing equivalences: https://github.com/Qiskit/qiskit-terra/issues/9511

**Pull request merged**